### PR TITLE
error out in unusual normalize conditions

### PR DESCRIPTION
**Pull request by heavycrystal**
primarily for CH

### CODE REVIEW

**Comment by serprex** [comment]
this'll happen when batches are empty due to pg_logical_emit_message

File: flow/connectors/clickhouse/normalize.go [hunk 1]
```diff
@@ -121,6 +124,13 @@ func generateCreateTableSQLForNormalizedTable(
 }
 
 func (c *ClickhouseConnector) NormalizeRecords(ctx context.Context, req *model.NormalizeRecordsRequest) (*model.NormalizeResponse, error) {
+	// ClickHouse is great, fix for potential consistency issues
+	select {
+	case <-time.After(3 * time.Second):
+	case <-ctx.Done():
+		return nil, ctx.Err()
+	}
```
Comment (serprex) [inline flow/connectors/clickhouse/normalize.go:132]: fine to sleep without select, let GetLastNormalizeBatchID find out ctx is done

**Review by serprex — APPROVED** [review]
want this tested, want Kaushik's approval

long term this ends up being weird logic which makes product more fragile

**Comment by heavycrystal** [comment]
flow will change a lot, can revisit this if needed

**Pull request closed without merging**